### ISSUE TITLE
exports is not defined

### ISSUE THREAD

**Issue by SunLxy** [web-infra-dev/rspack#3939]
### System Info

  System:
    OS: macOS 13.4.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 7.44 GB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.15.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.5.5 - /usr/local/bin/npm
    pnpm: 7.8.0 - /usr/local/bin/pnpm
    Watchman: 2023.04.17.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 115.0.5790.170
    Safari: 16.5.2
  npmPackages:
    @rspack/cli: 0.2.11 => 0.2.11

### Details

代码块中不能直接写 Object.defineProperty(exports, "__esModule", { value: true }) 

<img width="1894" alt="image" src="https://github.com/web-infra-dev/rspack/assets/49544090/206e5a2a-cb0b-423d-b74f-21e26bd02223">
<img width="1092" alt="image" src="https://github.com/web-infra-dev/rspack/assets/49544090/88ea66f6-daef-41f0-bf24-14c18ea73202">

### Reproduce Steps

1. npm install
2. npm run dev

**Comment by hardfist** [comment]
can you help check the behavior of webpack?

**Comment by underfin** [comment]
The webpack will render to 
```
var __webpack_unused_export__;

__webpack_unused_export__ = ({ value: true })
```

**Comment by SunLxy** [comment]
看了一下编译后的代码没有转换

<img width="1920" alt="image" src="https://github.com/web-infra-dev/rspack/assets/49544090/5cb63b49-affd-43be-b3f2-143da8be12a8">

**Comment by ScriptedAlchemy** [comment]
This is with an ESM output, correct?

**Comment by SunLxy** [comment]
这个就是初始化的项目，没加其他的配置 @ScriptedAlchemy 
<img width="990" alt="image" src="https://github.com/web-infra-dev/rspack/assets/49544090/04be3fd4-7c69-47ef-91eb-c2b925b732b9">

**Comment by daydayhappychao** [comment]
如果你的文件最终没有导出内容的话，构建工具会按 esm 处理，模块参数的第二项会是 webpack_exports 而不是 exports， webpack 也是这个逻辑，改一下你的 loader 中的模块化方案为 esm 就可以解决这个问题（当然为你的文件随便导出个啥让构建工具能知道你的文件类型是 commonjs 也可以

**Comment by ahabhgk** [comment]
This should already been fixed by #6883
